src/mantine-core/src/components/NumberInput/NumberInput.tsx

```diff
@@ -88,6 +94,8 @@ export const NumberInput = forwardRef<HTMLInputElement, NumberInputProps>(
       styles,
       size,
       rightSection,
+      formatter,
```

Comment:
Instead of conditionally applying formatter, create a default one and make it as default prop, same for parser

Reply:
@rtivital  that works for formatter, but what about parser? notice we `return` the parsed output: 
https://github.com/mantinedev/mantine/pull/774/files#diff-072fc6263e6e304bed6a5dbf188a6470c19309741aa4cc63462fa94ee01c84dcR147

Reply:
I do not see why this is an issue for having default parser, just restructure code to incorporate it